### DISCUSSION TITLE
Week 6 - Reading and Suggested ADA Tools

### DISCUSSION THREAD

**Discussion by bst3ck** [USU-ITLS-x265-23-summer/class-repo#130] [General]
## The Importance of ADA in our daily lives

A couple of years ago, while working at a nonprofit here in Salt Lake City, I was invited to attend a meeting with an individual who wanted to talk about incorporating some assistance for people who were blind into our new website. I was not interested in meeting with him at first but I was basically told that I needed to interview him to find out what type of requirements he would have for me. When looking at the already bulging backlog of tickets that needed to be addressed, I was willing to hear him out but I had no plans to actually do any of those tasks.

When I walked into the room that I was going to use for interviewing him. I saw a man sitting a chair with his computer in front of him, but is was obvious that he was blind. As I sat down and he introduced himself, I realized why he wanted to be involved. He went on to explain how hard it was for him to access our website and do the tasks necessary. I watched as he tried to give verbal commands using a program on his computer. He tried to use a screen reader to read out the content of our pages, but it would often get hung up and skipped large portions of the page.

I was embarrassed with how poorly his experience was on our site. I had rationalized that no one really cared about our site enough to try and use it with accessibility software. I felt that the small margin of users that fell into this category would not really mind not using our site, but I was wrong. That meeting changed the way I viewed how our team developed applications.

As part of my current team's development "Definition of Done" is the need to review all the standards listed in the WCAG 2.0 AA standards to ensure that the code they developed would meet those standards. It is surprising how this has changed how my developers and designers plan new components and templates. Each team member has become an advocate for good accessible designs. I know my team still has a lot of work to do, but our new websites that are launching over the next few months have had accessibility as one of our main goals.

## Great Accessibility Tools available for designers and developers

It may not be widely known, but Utah State University has an accessibility consulting group that is used by many organizations around the United States (and maybe farther). The service is called [WebAim](https://webaim.org/). Not only does WebAIM provide details on how to become certified, but it also has tools, checklists, and evaluation guides available on its website.

We have contracted with WebAim a few times over the last few years to come in and conduct training for our development staff on what they can do to be more accessibility minded in their work. They have instructed on detailed specifications on how to make specific features screen reader friendly, keyboard navigable, and designs that do not violate readability standards.

The best tool that I use almost every day is the WebAim Google Chrome plugin called [WAVE](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh?hl=en-US). Once this tool is installed on Chrome, it will display a summary of your accessibility results including mark up on the page that highlights items found. It is not 100% perfect and there will always be some sort of human evaluation needed, but it shows you what it thinks may be a problem and what part of the WCAG standards it may be violating. I would highly encourage the use of this tool. It is also fun to see it on other very popular websites to see how they rank with each other.

![image](https://github.com/USU-ITLS-x265-23-summer/class-repo/assets/13371999/b998219a-5aa6-4975-b247-5e07cb21cd89)

**Comment by metasean** [comment]
> As part of my current team's development "Definition of Done" is the need to review all the standards listed in the WCAG 2.0 AA standards to ensure that the code they developed would meet those standards. It is surprising how this has changed how my developers and designers plan new components and templates. Each team member has become an advocate for good accessible designs.

I'm not blind, but I have some visual challenges, and you'd be surprised at how many websites don't even meet my needs.  Your team may appreciate these [accessibility posters](https://accessibility.blog.gov.uk/2016/09/02/dos-and-donts-on-designing-for-accessibility/).

>I know my team still has a lot of work to do, but our new websites that are launching over the next few months have had accessibility as one of our main goals.

💟 

WebAim and Wave are actually pretty well known and well used through the web development community, by anyone who pays attention to accessibility.  And yes, they are great resources.  WebAim also does trainings, certifications and a few other things.  I was actually using their resources several years before I realized they were out of USU.  I was saddened that I hadn't even known they existed when I was a student there.

fwiw: I'll actually be using WAVE to assess the accessibility of your projects. 😁 So you're already ahead of the curve on this one!
